modules/dns_common/dns_common.rb

```diff
@@ -8,16 +9,20 @@ class Collision < RuntimeError; end
 
   class Record
     include ::Proxy::Log
+    include ::Proxy::TimeUtils
 
     attr_reader :server, :ttl
 
-    def initialize(server = nil, ttl = nil)
+    def initialize(server = nil, ttl = nil, resolv_timeout = nil)
       @server = server || "localhost"
-      @ttl    = ttl || "86400"
+      @ttl = ttl || "86400"
+      @resolv_timeout = resolv_timeout || 10
     end
```

Comment:
Optionally timeout can be now specified, Ruby default is quite tricky to understand.